src/router/bundle/bundleGenerator.ts

```diff
@@ -129,4 +129,56 @@ export default class BundleGenerator {
         response.entry = entries;
         return response;
     }
+
+    static generateBatchBundle(baseUrl: string, bundleEntryResponses: BatchReadWriteResponse[]) {
+        const id = uuidv4();
+        const response = {
+            resourceType: 'Bundle',
+            id,
+            type: 'batch-response',
+            link: [
+                {
+                    relation: 'self',
+                    url: baseUrl,
+                },
+            ],
+            entry: [],
+        };
+
+        const entries: any = [];
+        bundleEntryResponses.forEach((bundleEntryResponse) => {
+            let status = '200 OK';
+            if (bundleEntryResponse.error) {
+                status = bundleEntryResponse.error;
+            } else if (bundleEntryResponse.operation === 'create') {
+                status = '201 Created';
+            } else if (
+                ['read', 'vread'].includes(bundleEntryResponse.operation) &&
+                isEmpty(bundleEntryResponse.resource)
+            ) {
+                status = '403 Forbidden';
+            }
+            const entry: any = {
+                response: {
+                    status,
+                    location: `${bundleEntryResponse.resourceType}/${bundleEntryResponse.id}`,
+                    etag: bundleEntryResponse.vid,
+                    lastModified: bundleEntryResponse.lastModified,
+                },
+            };
+            // remove unnecessary fields from response
+            if (bundleEntryResponse.error) {
```

Comment:
Put this logic in Persistence instead of here? If etag and lastModified should not exist, it is Persistence's responsibility to remove them instead of depending on another package to handle it.

Reply:
I don't think it's possible to remove the vid/lastmodified fields from the persistence response as they are a required property of the type. However, I removed the logic to explicitly remove them, and instead combined the logic with the transaction bundle generation.